drivers/wifi/nrf700x/zephyr/src/qspi/src/ficr_prog.c

```diff
@@ -0,0 +1,452 @@
+	write_word(OTP_TIMING_REG1_ADDR, val);
+
+	val = OTP_TIMING_REG2_VAL;
+	write_word(OTP_TIMING_REG2_ADDR, val);
+}
+
+static int poll_otp_ready(void)
+{
+
+	int otp_mem_status = 0;
+	int poll = 0;
+
+	while ((otp_mem_status != OTP_READY) && (poll != 100)) {
+		read_word(OTP_POLL_ADDR, &otp_mem_status);
+
+		if ((otp_mem_status&0x4) == OTP_READY) {
+			return 0;
+		}
+		poll += 1;
+	}
+
+	LOG_INF("OTP is not ready\n");
+	return -ENOEXEC;
+}
+
+
+static int req_otp_standby_mode(void)
+{
+
+	uint32_t val;
+	int32_t otp_ready_exception;
+
+	val = 0x0;
+	write_word(OTP_RWSBMODE_ADDR, val);
+	otp_ready_exception = poll_otp_ready();
+	return otp_ready_exception;
+}
+
+
+static int otp_wr_voltage_2V5(void)
+{
+	uint32_t val = OTP_VOLTCTRL_2V5;
+	int32_t err;
+
+	err = req_otp_standby_mode();
+
+	if (err) {
+		LOG_INF("Failed Setting OTP voltage IOVDD to 2.5V\n");
+		return -ENOEXEC;
+	}
+	write_word(OTP_VOLTCTRL_ADDR, val);
+	return 0;
+}
+
+static int poll_otp_read_valid(void)
+{
+	int otp_mem_status = 0;
+	int poll = 0;
+
+	while ((otp_mem_status != OTP_READ_VALID) && (poll < 100)) {
+		read_word(OTP_POLL_ADDR, &otp_mem_status);
+
+		if ((otp_mem_status&0x2) == OTP_READ_VALID) {
+			return 0;
+		}
+		poll += 1;
+	}
+	LOG_INF("failed poll_otp_read_valid()\n");
+	return -ENOEXEC;
+}
+
+static int poll_otp_wrdone(void)
+{
+	int otp_mem_status = 0;
+	int poll = 0;
+
+	while ((otp_mem_status != OTP_WR_DONE) && (poll < 100)) {
+		read_word(OTP_POLL_ADDR, &otp_mem_status);
+
+		if ((otp_mem_status&0x1) == OTP_WR_DONE) {
+			return 0;
+		}
+		poll += 1;
+	}
+
+	LOG_INF("failed poll_otp_wrdone()\n");
+	return -ENOEXEC;
+}
+
+static int req_otp_read_mode(void)
+{
+	uint32_t val = OTP_READ_MODE;
+	int32_t err;
+
+	write_word(OTP_RWSBMODE_ADDR, val);
+	err = poll_otp_ready();
+	return err;
+}
+
+
+static int req_otp_byte_write_mode(void)
+{
+	uint32_t val = OTP_BYTE_WRITE_MODE;
+	int32_t err;
+
+	write_word(OTP_RWSBMODE_ADDR,  val);
+	err = poll_otp_ready();
+
+	if (err) {
+		return err;
+	}
+	return 0;
+
+}
+
+static uint32_t read_otp_location_disp(uint32_t offset, uint32_t *read_val)
+{
+
+	uint32_t addr = OTP_RDENABLE_ADDR;
+	uint32_t rd_addr = OTP_READREG_ADDR;
+	int32_t err;
+
+	write_word(addr,  offset);
+	err = poll_otp_read_valid();
+	if (err) {
+		LOG_INF("OTP read failed\n");
+		return err;
+	}
+	read_word(rd_addr, read_val);
+	return *read_val;
+
+}
+
+static int  write_otp_location(uint32_t otp_location_offset, uint32_t otp_data)
+{
+	int ret;
+
+	write_word(OTP_WRENABLE_ADDR,  otp_location_offset);
+
+	write_word(OTP_WRITEREG_ADDR,  otp_data);
+
+	ret = poll_otp_wrdone();
+
+	return ret;
+}
+
+
+static int otp_rd_voltage_1V8(void)
+{
+	uint32_t val = OTP_VOLTCTRL_1V8;
+	int32_t err;
+
+	err = req_otp_standby_mode();
+
+	if (err) {
+		LOG_INF("error in %s\n", __func__);
+		return -ENOEXEC;
+	}
+	write_word(OTP_VOLTCTRL_ADDR, val);
+	return 0;
+}
+
+
+int write_otp_memory(uint32_t otp_addr, uint32_t *write_val)
+{
+	int32_t		err;
+	int32_t		mask_val;
+
```

Comment:
For consistency, keep only 1 space between data type and variable name.

Reply:
Done